### PR TITLE
Check sorting of BAM file.

### PR DESCRIPTION
**Pull request by joshuak94**
Introduces a check to check that the BAM file is properly sorted. Address issue #86

### CODE REVIEW

File: test/data/simulated.minimap2.hg19.coordsorted_cutoff.sam [hunk 1]
```diff
@@ -1,3 +1,5 @@
+@HD	VN:1.6	SO:coordinate
+@SQ	SN:chr21	LN:46709983
```
Comment (Irallia) [inline test/data/simulated.minimap2.hg19.coordsorted_cutoff.sam:2]: If you make changes in our test data you have to update the shasum in the datasources.cmake

**Review by Irallia — CHANGES_REQUESTED** [review]
Nice and short! Could you write a short test with a mooked unsorted sam?

The failed test, comes from the changed shasum. ;)

File: test/data/simulated.minimap2.hg19.coordsorted_cutoff.sam (hunk 1 above)
Comment (joshuak94) [reply test/data/simulated.minimap2.hg19.coordsorted_cutoff.sam:2]: Done!

File: test/api/junction_detection_test.cpp [hunk 2]
```diff
@@ -65,6 +66,31 @@ TEST(junction_detection, fasta_out_not_empty)
...
+    unsorted_sam.close();
+
+    EXPECT_DEATH(detect_variants_in_alignment_file(empty_alignment_short_reads_file_path,
+                                      unsorted_sam_path,
+                                      tmp_dir/"detect_breakends_out_short.fasta",
+                                      default_methods,
+                                      simple_clustering,
+                                      no_refinement,
+                                      sv_default_length,
+                                      empty_output_path), "");
```
Comment (Irallia) [inline test/api/junction_detection_test.cpp:88]:
```suggestion
    EXPECT_DEATH(detect_variants_in_alignment_file(empty_alignment_short_reads_file_path,
                                                   unsorted_sam_path,
                                                   tmp_dir/"detect_breakends_out_short.fasta",
                                                   default_methods,
                                                   simple_clustering,
                                                   no_refinement,
                                                   sv_default_length,
                                                   empty_output_path), "");
```

File: test/api/junction_detection_test.cpp [hunk 3]
```diff
@@ -65,6 +66,31 @@ TEST(junction_detection, fasta_out_not_empty)
...
+
+    std::filesystem::remove(tmp_dir/"detect_breakends_out_short.fasta");
+    std::filesystem::remove(unsorted_sam_path);
+
```
Comment (Irallia) [inline test/api/junction_detection_test.cpp:92]: new line
```suggestion
```

**Review by Irallia — APPROVED** [review]
Wonderful, thank you very much!
I still found a style issue, but apart from that excellent. 💅

**Comment by joshuak94** [comment]
@Irallia Updated and cleaned up commits.

File: src/variant_detection/variant_detection.cpp [hunk 4]
```diff
@@ -33,6 +33,12 @@ void detect_junctions_in_long_reads_sam_file(std::vector<Junction> & junctions,
                                      seqan3::field::header_ptr>;
     seqan3::sam_file_input alignment_long_reads_file{alignment_long_reads_file_path, my_fields{}};
 
+    // Check that the file is sorted before proceeding.
+    if (alignment_long_reads_file.header().sorting != "coordinate")
+    {
+        seqan3::debug_stream << "ERROR: Input file must be sorted by coordinate (e.g. samtools sort)" << std::endl;
```
Comment (MitraDarja) [inline src/variant_detection/variant_detection.cpp:39]: Why are you not using a throw?

**Review by MitraDarja — CHANGES_REQUESTED** [review]
LGTM, but one question:

**Review by MitraDarja — CHANGES_REQUESTED** [review]

**Review by MitraDarja — APPROVED** [review]

**Comment by MitraDarja** [comment]
@Irallia I think, you can merge now. :)

**Pull request merged**